### PR TITLE
Fix for Xml variable substitution

### PR DESCRIPTION
**Pull request by Ajay-MS**
**Scenario**
Web App package contains web.config and parameters.xml and then xml variable substitution on config section will fail. 

**Cause**
LtxDomUtility class had xmlDomLookUpTable  as global variable and when  initializing  xmlDom for parameters.xml it's overriding lookupUpTable for web.config.

**Fix**
Changed LtxDomUtility to a class. So each instance of class will have it's own lookup table.

### CODE REVIEW

**Comment by shaneholder** [comment]
I can confirm that using the instance of the lxdomutility does appear to resolve my current issue.  However, it does seem to be a bit overkill for the number of times it is actually used.  But it does resolve my current issue.

File: Tasks/Common/webdeployment-common/ltxdomutility.ts [hunk 1]
```diff
@@ -2,89 +2,96 @@ var ltx = require("ltx");
...
 
-export function initializeDOM(xmlContent) {
-    xmlDomLookUpTable = {};
-    headerContent = null;
-    var xmlDom = ltx.parse(xmlContent);
-    readHeader(xmlContent);
-    buildLookUpTable(xmlDom);
-    return xmlDom;
-}
+    private xmlDomLookUpTable = {};
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/ltxdomutility.ts]: Can we assign type to each variable?

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "version": {
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/task.json]: patch IIS Web app deploy task too

File: Tasks/Common/webdeployment-common/ltxdomutility.ts [hunk 3]
```diff
@@ -2,89 +2,96 @@ var ltx = require("ltx");
...
+    }
+
+    public getContentWithHeader(xmlDom) {
+        return xmlDom ? (this.headerContent ? this.headerContent + "\n" : "") + xmlDom.root().toString() : "";
+    }
+
+    /**
+     * Define method to create a lookup for DOM 
+     */
+    public buildLookUpTable(node) {
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/ltxdomutility.ts]: make it private

File: Tasks/Common/webdeployment-common/ltxdomutility.ts [hunk 4]
```diff
@@ -2,89 +2,96 @@ var ltx = require("ltx");
...
-function buildLookUpTable(node) {
-    if(node){
-        var nodeName = node.name;
-        if(nodeName){
-            nodeName = nodeName.toLowerCase();
-            var listOfNodes = xmlDomLookUpTable[nodeName];
-            if(listOfNodes == null || !(Array.isArray(listOfNodes))) {
-                listOfNodes = [];
-                xmlDomLookUpTable[nodeName] = listOfNodes;
+    public readHeader(xmlContent) {
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/ltxdomutility.ts]: make it private

File: Tasks/AzureRmWebAppDeployment/task.json (hunk 2 above)
Comment (Ajay-MS) [reply Tasks/AzureRmWebAppDeployment/task.json]: Done

File: Tasks/Common/webdeployment-common/ltxdomutility.ts (hunk 1 above)
Comment (Ajay-MS) [reply Tasks/Common/webdeployment-common/ltxdomutility.ts]: To keep it simple skipping it as we have custom types belonging to different modules.

**Comment by Ajay-MS** [comment]
@shaneholder 
Can you please briefly describe 'bit overkill for the number of times it is actually used' ?

As per my understanding earlier we were creating single instance now there will be two instances.  Each variable scope will be within that iteration of for loop.

**Pull request merged**